### PR TITLE
Add github-stats tool

### PR DESCRIPTION
**Pull request by adamwalach**
**Description**

Changes proposed in this pull request:

- created tool for gathering statistics from github (types of issues)

**Related issue(s)**
https://github.com/kyma-project/kyma/issues/5476

### CODE REVIEW

**Comment by adamwalach** [comment]
Logs for manually executed job: https://status.build.kyma-project.io/log?job=github-stats&id=1204027989390528512

File: development/tools/cmd/githubstats/README.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    --github-access-token={GitHub OAuth2 access token} \
+    --github-repo-name={GitHub repository name} \
+    --github-repo-owner={GitHub repository owner}
+```
+
+### Flags
+
+```bash 
+Usage:
+  githubstats [flags]
```
Comment (majakurcius) [inline development/tools/cmd/githubstats/README.md]:
``````suggestion
```
  githubstats [flags]
```
``````

File: development/tools/cmd/githubstats/README.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+Flags:
+  -t, --github-access-token string   GitHub token [Required] [APP_GITHUB_ACCESS_TOKEN]
```
Comment (majakurcius) [inline development/tools/cmd/githubstats/README.md]:
``````suggestion
```
  -t, --github-access-token string   GitHub token [Required] [APP_GITHUB_ACCESS_TOKEN]
``````

File: development/tools/cmd/githubstats/README.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
+# Github Statistics
```
Comment (majakurcius) [inline development/tools/cmd/githubstats/README.md]:
```suggestion
# GitHub Statistics
```

File: development/github-stats.sh [hunk 4]
```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env bash
+
+set -eu
+
+readonly DEVELOPMENT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+echo "--------------------------------------------------------------------------------"
+echo "Generating Github stats..."
```
Comment (majakurcius) [inline development/github-stats.sh]:
```suggestion
echo "Generating GitHub stats..."
```

File: development/tools/cmd/githubstats/main.go [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+	"github.com/spf13/cobra"
+	"golang.org/x/oauth2"
+)
+
+type Issues struct {
+	Open   IssuesOpen
+	Closed IssuesClosed
+}
+
+type IssuesOpen struct {
```
Comment (mszostok) [inline development/tools/cmd/githubstats/main.go]: why it's duplicated?

why u cannot have just 

```go
type Issues struct {
	Open   IssuesStats
	Closed IssuesStats
}
```

File: development/tools/cmd/githubstats/main.go [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+			} `graphql:"issues(states: [$state])"`
+		} `graphql:"repository(owner: $repositoryOwner, name: $repositoryName)"`
+	}
+)
+
+func getStats(cfg Config) {
+	src := oauth2.StaticTokenSource(
+		&oauth2.Token{AccessToken: cfg.GithubAccessToken},
+	)
+	httpClient := oauth2.NewClient(context.Background(), src)
```
Comment (mszostok) [inline development/tools/cmd/githubstats/main.go]: I'm not quite sure if you want but you can set up a graceful shutdown, so  `getStats` will accept a context param and this context will be closed when app will receive shutdown request (same for all bellow queries)

File: development/tools/cmd/githubstats/main.go [hunk 7]
```diff
@@ -0,0 +1,183 @@
...
+
+	variables := map[string]interface{}{
+		"repositoryOwner": githubv4.String(cfg.GithubRepoOwner),
+		"repositoryName":  githubv4.String(cfg.GithubRepoName),
+	}
+
+	// Total count
+	variables["state"] = githubv4.IssueState("OPEN")
+	err := client.Query(context.Background(), &query_no_labels, variables)
+	if err != nil {
```
Comment (mszostok) [inline development/tools/cmd/githubstats/main.go]: simply use fatalOnError:)

```suggestion
	fatalOnError(err)
```

File: development/tools/cmd/githubstats/main.go [hunk 8]
```diff
@@ -0,0 +1,183 @@
...
+		log.Fatal(err)
+	}
+	r.Issues.Open.TotalCount = query_no_labels.Repository.Issues.TotalCount
+
+	variables["state"] = githubv4.IssueState("CLOSED")
+	err = client.Query(context.Background(), &query_no_labels, variables)
+	if err != nil {
```
Comment (mszostok) [inline development/tools/cmd/githubstats/main.go]:
```suggestion
	fatalOnError(err)
```

File: development/tools/cmd/githubstats/main.go [hunk 9]
```diff
@@ -0,0 +1,183 @@
...
+	r.Issues.Open.PriorityCritical = executeQuery(client, variables)
+
+	json, err := json.Marshal(r)
+	fatalOnError(err, "while marshaling json")
+	fmt.Println(string(json))
+}
+
+func executeQuery(client *githubv4.Client, variables map[string]interface{}) int64 {
+	err := client.Query(context.Background(), &query, variables)
+	if err != nil {
```
Comment (mszostok) [inline development/tools/cmd/githubstats/main.go]: you do not need the `if` statement here

File: development/tools/cmd/githubstats/main.go [hunk 10]
```diff
@@ -0,0 +1,183 @@
...
+	rootCmd.PersistentFlags().StringVarP(&cfg.GithubRepoOwner, "github-repo-owner", "o", "", "owner/organisation name [Required]")
+	rootCmd.PersistentFlags().StringVarP(&cfg.GithubRepoName, "github-repo-name", "r", "", "repository name [Required]")
+
+	rootCmd.MarkPersistentFlagRequired("github-token")
+	rootCmd.MarkPersistentFlagRequired("owner")
+	rootCmd.MarkPersistentFlagRequired("repo")
+
+	envy.ParseCobra(rootCmd, envy.CobraConfig{Prefix: "APP", Persistent: true, Recursive: false})
+
+	if err := rootCmd.Execute(); err != nil {
```
Comment (mszostok) [inline development/tools/cmd/githubstats/main.go]: use just `fatalOnError`

File: docs/prow/prow-jobs.md [hunk 11]
```diff
@@ -95,9 +95,11 @@ go run development/tools/cmd/rendertemplates/main.go --config templates/config.y
 In most situations, re-running the job means that Prow uses the same commit. To make sure the job uses the updated code base, you can trigger the job manually.
 Use a tool called [mkpj](https://github.com/kubernetes/test-infra/tree/master/prow/cmd/mkpj) that generates a valid YAML for the Prow job. Follow these steps:
 
-1. Clone the `kubernetes/test-infra` repository. Run:
+1. Clone the `kubernetes/test-infra` repository into `k8s.io` directory. Run:
 
 ```shell
+mkdir -p $GOPATH/src/k8s.io/
+cd $GOPATH/src/k8s.io
 git clone "git@github.com:kubernetes/test-infra.git"
```
Comment (mszostok) [inline docs/prow/prow-jobs.md]: FYI: you can make it as one-liner:

```shell
git clone "git@github.com:kubernetes/test-infra.git" $GOPATH/src/k8s.io
```

this command will take care of creating the necessary folders

File: development/tools/cmd/githubstats/main.go (hunk 5 above)
Comment (adamwalach) [reply development/tools/cmd/githubstats/main.go]: Right, I somehow forgot about that

File: development/tools/cmd/githubstats/main.go (hunk 10 above)
Comment (adamwalach) [reply development/tools/cmd/githubstats/main.go]: I think in this case I could use just os.Exit(), from what I can see Execute() already prints error to stderr
With os.Exit:
```
Error: required flag(s) "github-repo-name", "github-repo-owner" not set
Usage:
  githubstats [flags]

Flags:
  -t, --github-access-token string   github token [Required] [APP_GITHUB_ACCESS_TOKEN]
  -r, --github-repo-name string      repository name [Required] [APP_GITHUB_REPO_NAME]
  -o, --github-repo-owner string     owner/organisation name [Required] [APP_GITHUB_REPO_OWNER]
  -h, --help                         help for githubstats

exit status 1
```
with fatalOnError:
```
Error: required flag(s) "github-repo-name", "github-repo-owner" not set
Usage:
  githubstats [flags]

Flags:
  -t, --github-access-token string   github token [Required] [APP_GITHUB_ACCESS_TOKEN]
  -r, --github-repo-name string      repository name [Required] [APP_GITHUB_REPO_NAME]
  -o, --github-repo-owner string     owner/organisation name [Required] [APP_GITHUB_REPO_OWNER]
  -h, --help                         help for githubstats

2019/12/09 19:37:02 : required flag(s) "github-repo-name", "github-repo-owner" not set
```
the error message is repeated

File: docs/prow/prow-jobs.md (hunk 11 above)
Comment (adamwalach) [reply docs/prow/prow-jobs.md]: nice :)

File: development/tools/cmd/githubstats/README.md [hunk 12]
```diff
@@ -0,0 +1,72 @@
...
+go run main.go \ 
+    --github-access-token={GitHub OAuth2 access token} \
+    --github-repo-name={GitHub repository name} \
+    --github-repo-owner={GitHub repository owner}
+```
+
+### Flags
+
+Usage:
+```
```
Comment (majakurcius) [inline development/tools/cmd/githubstats/README.md]:
```suggestion
```bash
```

File: development/tools/cmd/githubstats/README.md [hunk 13]
```diff
@@ -0,0 +1,72 @@
...
+  githubstats [flags]
+```
+
+Flags:
+```
```
Comment (majakurcius) [inline development/tools/cmd/githubstats/README.md]:
```suggestion
```bash
```

File: development/tools/cmd/githubstats/main.go (hunk 6 above)
Comment (adamwalach) [reply development/tools/cmd/githubstats/main.go]: Good idea, fixed

**Pull request merged**